module/shared-kernel/src/Domain/AbstractCode.php

```diff
@@ -1,29 +1,28 @@
 <?php
-
-/**
+/*
  * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
  * See LICENSE.txt for license details.
  */
 
 declare(strict_types=1);
 
-namespace Ergonode\Core\Domain\Entity;
+namespace Ergonode\SharedKernel\Domain;
 
 use JMS\Serializer\Annotation as JMS;
 
-abstract class AbstractCode
+class AbstractCode
 {
     public const MIN_LENGTH = 1;
-    public const MAX_LENGTH = 255;
-    public const PATTERN = '/^[a-zA-Z0-9-_ ]+$\b/i';
+    public const MAX_LENGTH = 128;
 
     /**
      * @JMS\Type("string")
      */
-    private string $value;
+    protected string $value;
 
     public function __construct(string $value)
     {
+        $value = trim($value);
         if (!self::isValid($value)) {
```

Comment:
If you'd use static call you wouldn't have to execute validation multiple times in some of the implementations but just override the method.